eos_data_distribution/ndn/tests/test_file.py

```diff
@@ -137,6 +142,316 @@ def test_bitmap_to_num_invalid(self):
             file.bitmap_to_num([0, 0, 0, 0, 0, 0, 0, 2])
 
 
+class MockFace(object):
+
+    """
+    Mock Face implementation for the purposes of testing base classes.
+
+    This is a mock implementation of ``pyndn.face.Face`` which does no network
+    communications, but looks like a ``Face`` implementation through duck
+    typing.
+
+    It records the interests expressed to it, and allows the test harness to
+    reply to those interests with data (``callInterestDone()``) or a timeout
+    (``callInterestTimeout()``).
+
+    It exposes two sets of API: the public API for ``Face``, and the test
+    harness API.
+    """
+
+    def __init__(self):
+        # dict mapping name to (Interest, onData, onTimeout) tuple
+        self._interests = {}
+
+    # Public API to emulate Face.
+
+    def removePendingInterest(self, name):
+        raise NotImplementedError('Not implemented in MockFace yet')
+
+    @property
+    def usesGLibMainContext(self):
+        return True
+
+    def setCommandSigningInfo(self, key_chain, certificate_name):
+        raise NotImplementedError('Not implemented in MockFace yet')
+
+    def expressInterest(self, interest, on_data, on_timeout):
+        self._interests[str(interest.name)] = (interest, on_data, on_timeout)
```

Comment:
Why is that better than accessing it as a property (which is a bit less typing)? The documentation makes them look equivalent: http://named-data.net/doc/0.4.0/PyNDN2/pyndn.html#pyndn.interest.Interest